test/test_nn.py

```diff
@@ -12067,6 +12067,76 @@ def test_threshold_inplace_overlap(self, device):
         F.threshold(x, 0.5, 0.5, inplace=True)
         F.threshold_(x, 0.5, 0.5)
 
+    def test_triplet_margin_loss_with_distance_default_parity(self, device):
```

Comment:
Have the additional parameters of margin, swap, and reduction vary, too. You could do something like:

```
for extra_args in product((.5, 1, 1.25), (True, False), (True, False)):
  kwargs = {'margin': extra_args[0], ... }

  # Tests forward
  expected = F.triplet_margin_loss(anchor, positive, negative, **kwargs)
  actual = F.triplet_margin_loss_with_distance(anchor, positive, negative, **kwargs)
  self.assertEqual(actual, expected, rtol=0, atol=0)

  # Tests backward
  ...
```

For the backward you may have to bind the kwargs using `partial` before calling to `gradcheck`.

Reply:
I think I'm essentially doing this with the next test; this one is checking that the default args are aligned.

Reply:
Not sure if the callable iterator would work, since the module takes the kwargs at construction-time and the functional takes them at call-time...